src/qat/purr/qatconfig.py

```diff
@@ -86,10 +87,16 @@ def check_disable_pulse_duration_limits(cls, DISABLE_PULSE_DURATION_LIMITS):
             )
         return DISABLE_PULSE_DURATION_LIMITS
 
+    def validate(self, compiler_config: CompilerConfig):
```

Comment:
I think this should be called validate_compiler_config just to make it clear what it's validating?